flink-cdc-connect/flink-cdc-source-connectors/flink-connector-debezium/src/main/java/org/apache/flink/cdc/debezium/internal/DebeziumChangeFetcher.java

```diff
@@ -248,8 +248,8 @@ private void handleBatch(List<ChangeEvent<SourceRecord, SourceRecord>> changeEve
                 LOG.error("Failed to deserialize record {}", record, t);
                 throw t;
             }
-
-            if (isInDbSnapshotPhase && !isSnapshotRecord(record)) {
+            // Wait for the end of the snapshot phase for the first data to trigger subsequent calculations.
+            if (isInDbSnapshotPhase && !isSnapshotRecord(record) && messageTimestamp !=0) {
```

Comment:
Please make sure that `mvn spotless:apply` / `mvn spotless:check` have been executed before committing to resolve any codestyle violations